### ISSUE TITLE
Allow suites to override the run directory location?

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc#390]
I have one user who wants to be able to put all suite and job logs, and share and work directories etc. inside his suite definition directories. This used to be possible pre site/user config file when all these directories were individually configurable in runtime namespaces. Personally I'm not fond of the idea, but I'm posting this to flag it for consideration... It could be achieved in several ways, e.g.:
 * a "run directory" item in the suite.rc which if used would override the site/user item.
 * a single "put suite output in suite definition directory" option in the site/user file.
 * suite-specific variables in site/user "run directory" (requiring the suite definition to be parsed before site/user settings)

Implications for other task hosts would have to be thought through. And: non-suite access to the suite would require parsing the suite definition again. 

Low priority at best as users can use symbolic links in the suite definition directory to get essentially the same effect now.

**Comment by hjoliver** [comment]
Closing this.   No one can think of a good reason to not use the standard locations, especially after #1885, when data directories can be redirected with symlinks as required.
